pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -340,6 +330,23 @@ func (ssc *defaultStatefulSetControl) updateStatefulSet(
 		// If the ordinal could not be parsed (ord < 0), ignore the Pod.
 	}
 
+	// make sure to update the latest status even if there is an error later
+	defer func() {
```

Comment:
I have similar concerns wrt this approach outlined here: https://github.com/kubernetes/kubernetes/pull/112127#discussion_r979925952 before we start any backports I'd prefer we fix it the right way.